### PR TITLE
Bundle standalone using rollup

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Shrink `*-standalone` size
| Tests Added + Pass?      | Nope
| Documentation PR Link    | 
| Any Dependency Changes?  | Rollup plugins are added, webpack plugins/gulp-uglify are removed.
| License                  | MIT

This PR supersedes #7581 and is loosely based on #7581 because I find it too difficult to rebase.

This PR translates the webpack config in the language of rollup config. `gulp-uglify` is also replaced by `rollup-plugin-terser`.

Updates:
There is a fatal error on the previous size comparison table, some of the dependencies of `babel-preset-env-standalone` was _not_ bundled due to the `importer` is not an ES Module. The new size comparison has reflected on this fix. Thanks to @nicolo-ribaudo who fist discovered this error.

Here is the size comparison between [Circle CI Artifacts](https://app.circleci.com/jobs/github/babel/babel/13105) and `*-standalone` 7.7.4 ([babel](https://unpkg.com/browse/@babel/standalone@7.7.4/), [env](https://unpkg.com/browse/@babel/preset-env-standalone@7.7.4/))

| | **Previous Size** | **New Size** | **Change** |
| --- | --- | --- | --- |
| babel.js | 3.04 MB | 2.6 MB | -14% |
| babel.min.js | 1.63 MB | 1.2 MB | -26% |
| babel-preset-env.js | 2.93 MB | 3.3 MB | +13% |
| babel-preset-env.min.js | 1.7 MB | 1.3 MB | -24% |

The increased size in `babel-preset-env.js` is from the fact that rollup does not strip the comments by default. For production usage, please refer to `babel-preset-env.min.js`.

<s>
Here is the size comparison between [Circle CI Artifacts](https://app.circleci.com/jobs/github/babel/babel/13068) and `*-standalone` 7.7.4 ([babel](https://unpkg.com/browse/@babel/standalone@7.7.4/), [env](https://unpkg.com/browse/@babel/preset-env-standalone@7.7.4/))

| | **Previous Size** | **New Size** | **Change** |
| --- | --- | --- | --- |
| babel.js | 3.04 MB | 2.6 MB | 14% |
| babel.min.js | 1.63 MB | 1.2 MB | 26% |
| babel-preset-env.js | 2.93 MB | 1.0 MB | 66% |
| babel-preset-env.min.js | 1.7 MB | 450 kB | 74% |
</s>

(It seems that `buildsize` does not comment for unknown reasons...)

### CODE REVIEW

File: rollup-plugin-babel-source.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+const path = require("path");
```
[the remark this answers is not in the document]
Comment (JLHwung) [reply rollup-plugin-babel-source.js]: Sure I forgot about that.

File: babel.config.js [hunk 2]
```diff
@@ -126,21 +114,11 @@ module.exports = function(api) {
           "codemods/*/test",
           "eslint/*/src",
           "eslint/*/test",
+          "**/node_modules",
+          "packages/babel-preset-env/data",
         ],
         sourceType: "unambiguous",
       },
-      {
```
Comment (JLHwung) [inline babel.config.js]: They are removed since `rollup-plugin-babel` can combine the helpers into a single block.

File: babel.config.js [hunk 3]
```diff
@@ -51,23 +49,13 @@ module.exports = function(api) {
...
-    };
-  }
-
   const config = {
     // Our dependencies are all standard CommonJS, along with all sorts of
     // other random files in Babel's codebase, so we use script as the default,
     // and then mark actual modules as modules farther down.
     sourceType: "script",
     comments: false,
+    exclude: "**/node_modules/**",
```
Comment (nicolo-ribaudo) [inline babel.config.js]: What is the reason for this change? The comment above implies that this config is mainly meant for `node_modules`.

Comment (JLHwung) [reply babel.config.js]: Oh good catch! It was meant for to deal with `'import' and 'export' may appear only with 'sourceType: "module"'` since rollup commonjs converts them into ESM, but apparently it is an overkill.

**Comment by nicolo-ribaudo** [comment]
CircleCI is failing because of core-js 3.4.6. It's fixed in 3.4.7, but I don't know how to clear the cache before triggering a build.
